### ISSUE TITLE
perf: Indefinite loss of interactivity

### ISSUE THREAD

**Issue by wangandi** [MaterializeInc/materialize#4771]
Likely a subissue of #1537.

I managed to produce twice using the linux release tarball version (commit id a8d87436) where the loss of interactivity lasts until Materialize is turned off. I used gdb to see what was going on, but gdb showed that Materialize was responding to queries fine. Indeed, if you ping "/status" before and after firing a query, you will see that the number in "mz_command_durations command=query status=success <number>" increments even though on the client end, the query never returns. Top shows that Materialize is only using a tenth of available memory and CPU.

This issue is not actionable unless somebody can reliably reproduce the issue with a dataset that can be shared across the org. The indefinite loss of interactivity seems to be associated with deleting an index (~18GB), quickly recreating it (or an equally large index), and then immediately trying to query it.

**Comment by wangandi** [comment]
I can actually reproduce an indefinite loss of interactivity relatively reliably in both debug and release mode using 1 GB of TPCH data. Though in this case of indefinite loss of interactivity, the number of successful queries does not increment if you ping "/status"

How to do this:
1) Don't materialize any of the sources, but create a materialized view for any of the queries.
2) Drop the materialized view above. You don't have to drop it within a particular amount of time after creating the view; I have reproduced the issue dropping the materialized view after taking an hour-long lunch break.
3) Create a materialized view for a different query relatively quickly after dropping the previous one (the window appears to be at least 30 seconds long, possibly even several minutes long. Same query might also work, but I did not test it.)
4) Immediately start pinging "/memory" from your browser after creating the second materialized view.
